XIPs/xip-10-brandinfo-schema.md

```diff
@@ -0,0 +1,90 @@
+---
+xip: 14
+title: Conversation context metadta schema
+description: Schema to enable standardized sharing of conversation context
+author: Yash Lunagaria (@yash-luna)
+status: Review
+type: Standards Track
+category: XRC
+created: 2023-01-31
+---
+
+## Abstract
+
+This XIP proposes a schema for sharing additional context about a conversation beyond a `conversationId`. Having a standard way to provide not only a conversationId, but also conversation metadata will enable different frontends to display useful contextual information in a consistent, user-friendly manner, while still affording apps design and UI flexibility.
+
+## Motivation
+
+The SDK API currently accepts a unique `conversationId` per address pair and allows `metadata` to be set as well. Today, different apps use these parameters in different ways. Therefore an app may not know how to correctly render in its UI `conversationId` and `metadata` for conversations originating from other apps. We propose a schema to standardize the use of conversation `metadata` for the purpose of app brand expression stemming from the `conversationId`. The `metadata` field may include other non-standard (or future standard) properties.
+
+Note that the schema is only applicable when setting a non-null `conversationId`. If `conversationId` is null, the SDK does not allow setting `metadata`.
+
+## Specification
+
+Proposed `displayInfo` schema in conversation `metadata`
+
+```json
+{
+    conversationId: "mydomain.xyz/abc/qrs",
+    metadata: { 
```

Comment:
We may want to consider adding an explicit version to this schema, so future updates are easier to deploy.

The scheme could be directly tied to conversationVersion, But then schema changes would result in a bump in both. 

Perhaps something explicit like:
```
metadata: {
   "conversationSource_V1": {
       "prettyName": "My company",
       "profileImage": "mydomain.xyz/assets/myimage.png"
   }
}
```

That would allow the SDK to add multiple Objects to the metadata if older clients need to be supported

Reply:
@galligan  I like the explicit structure of version, type, content.

I might suggest a structure that looks like:

```
metadata: [
	{
		"type": "conversation_source",
		"version": "1.0" 
		"content": {
		  	"title": "My company",
    		"description"; "My description that may be used in an app",
    		"image": "https://mydomain.xyz/assets/myimage.png",
    		"imageURI": "ipfs://zafybeict2kq6gt4ikgulypt7h7nwj4hmfi2kevrqvnx2osibfulyy5x3hu/myimage.png"
		}
	},
	{
		"type": "group_info",
		"version": "1.0" 
		"content": {
		  	"name": "Frens",
    		"admin"; "....",
    		"image": "ipfs://pkrewapct2kq6gt4ikgulypt7h7nwj4hmfi2kevrqvnx2osibfulypkl39a/myimage.png".
		}
	}
]
```

- There should be room for multiple metadata objects
- Objects names ought to be hyper specific to the data it is representing. `Conversation.Image` is ambiguous. Codifying clear names for entities will help developers in the future.

Reply:
Worth noting that neither an array or nested object are possible with today's metadata. It is strictly a Map<string, string>. Closer to HTTP headers than a free form JavaScript object.

Reply:
Based on the suggestions above, there's benefits to the protocol and DX to enable metadata to take arrays and/or nested objects. It would enable versioning for metadata schemas and make it friendly for other use cases beyond conversationId context.

@neekolas @jazzz: What are the downsides of such a change? Can it be in scope for xmtp-js v8?